packages/trader/src/AppV2/Components/DraggableList/draggable-list.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useState } from 'react';
+import { DragDropContext, Droppable, Draggable, DropResult } from 'react-beautiful-dnd';
+import { Text } from '@deriv-com/quill-ui';
+import DraggableListItem from './draggable-list-item';
+
+export type DraggableListItem = {
+    id: string;
+    title: string;
+    icon?: React.ReactNode;
+}
+
+export type DraggableListCategory = {
+    id: string;
+    title?: string;
+    items: DraggableListItem[];
+}
+
+export type DraggableListProps = {
+    categories: DraggableListCategory[];
+    onRightIconClick: (item: DraggableListItem) => void;
+}
+
+const DraggableList: React.FC<DraggableListProps> = ({ categories, onRightIconClick }) => {
+    const [category_list, setCategoryList] = useState(categories);
+
+    const handleOnDragEnd = (result: DropResult) => {
+        if (!result.destination) return;
+
+        const source_category_index = category_list.findIndex(category => category.id === result.source.droppableId);
+        const dest_category_index = category_list.findIndex(
+            category => category.id === result.destination?.droppableId
+        );
+
+        if (source_category_index === -1 || dest_category_index === -1) return;
+
+        const source_category = category_list[source_category_index];
+        const dest_category = category_list[dest_category_index];
+
+        const source_items = Array.from(source_category.items);
+        const [moved_item] = source_items.splice(result.source.index, 1);
+
+        if (source_category_index === dest_category_index) {
+            source_items.splice(result.destination.index, 0, moved_item);
+            const new_category_list = Array.from(category_list);
+            new_category_list[source_category_index] = {
+                ...source_category,
+                items: source_items,
+            };
+            setCategoryList(new_category_list);
+        } else {
+            const dest_items = Array.from(dest_category.items);
+            dest_items.splice(result.destination.index, 0, moved_item);
+
+            const new_category_list = Array.from(category_list);
+            new_category_list[source_category_index] = {
+                ...source_category,
+                items: source_items,
+            };
```

Comment:
it seems like this part is exactly the same in both conditions and can be placed **before** `if-else`:
```
const new_category_list = [...category_list]; // replaced Array.from() with spread
new_category_list[source_category_index] = {
    ...source_category,
    items: source_items,
};
```
and this can be placed **after** `if-else`:
```
setCategoryList(new_category_list);
```
It might help make the condition smaller and more readable :)
Will it work?

Reply:
Sure, I made these changes